### ISSUE TITLE
user pemissions: citations

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#849]
The addition of citations to the identification form has caused an unexpected error: users without role MANAGE_PUBLICATIONS (and therefore no rights on table citation) cannot save Identifications when there's a citation.

**Should MANAGE_SPECIMENS be able to manage citations, or should that remain limited to MANAGE_PUBLICATIONS?**

We need an answer very quickly - this is stopping work.

If we want to be more liberal with citations, I can change the roles and the form will work. If not, I'm going to have to add a bunch of IF... code to the identifications form.

@DerekSikes @ccicero

**Comment by DerekSikes** [comment]
How hard would it be to have citations get their own little form inside the specimen form analogous to the edit ID form and the edit locality/event form and the edit attribute form, etc. Seems the most logical way to build access to edit citations.

**Comment by campmlc** [comment]
I think good students could manage publications, potentially, but would
they have access to projects as well? Has anyone had issues with giving
students this kind of access?

**Comment by dustymc** [comment]
>How hard would it be to have citations get their own little form inside the specimen form analogous to the edit ID form and the edit locality/event form and the edit attribute form, etc. Seems the most logical way to build access to edit citations.

That's how this started out, but as citations are linked to IDs (not specimens) it ended up being a LOT of code to maintain (pretty much everything from IDs) and in testing I ended up needing to make new IDs (new tab) and then hook citations to them (new tab) then notice something and edit the ID (new tab), then add a citation remark (new tab).... I think that, given the deep integration between IDs and citations (vs. parts, attributes, etc., which mostly stand alone) that this is a much better solution from a usability standpoint. The code is pretty complex, but it's not replicated so I think this is pretty OK from a maintenance standpoint as well.

>I think good students could manage publications....

Manage/Roles/DBRoles will tell you exactly what any given role can do. Publications and projects and citations are tangled up right now. I think the immediate question is whether citations are a specimen thing or a publication thing as far as roles go. (Reality is they're both, of course.)

See also #745 - **any** student who understands that publications are shared, knows how/why to use DOIs, etc., etc., etc. is probably just fine in there, while otherwise-amazing people who don't understand the implications of shared data are dangerous.

Creating roles is easy - this started out assigning everyone CRUD access on individual tables - but understanding/managing more than a few did not prove to be something that the average admin user was up for.

And I don't really care what the answer is - I can pull citations out of IDs for those users without access, or I can change the roles. I'd just rather not write form code if we decide we have the roles wrong - and given this new viewpoint to the data I think we may (but no strong opinions, I could go either way).

**Comment by campmlc** [comment]
I vote for giving Manage Specimens access to Manage Publications, if that is the easiest solution which will allow pending work to move forward. But perhaps whoever is doing that work can keep an eye on the situation for potential problems.

**Comment by ccicero** [comment]
I agree, I think that users with permission to Manage Specimens should also be able to manage citations.

**Comment by ccicero** [comment]
I am ok with keeping publications and projects together in one role, which probably should be restricted at some level (I wouldn't want students creating publications and projects), and pulling out 'manage citations' into 'manage specimens.' To me that's more of a specimen-based thing. However, I'm also ok with pulling citations out of IDS for users who don't have access. Users with more restricted roles probably shouldn't be updating identifications anyway.

**Comment by dustymc** [comment]
>keeping publications and projects together in one role

Yes, agreed, those are deeply integrated.

>probably should be restricted at some level

See #745 - currently anyone with a role and global_admin can grant that role to anyone they wish.

>Users with more restricted roles probably shouldn't be updating identifications anyway.

We could also make this finer-grained - manage_specimens can currently INSERT and UPDATE IDs but not DELETE them, for example, and we could do something similar here.

It seems like we're reaching a consensus on this, so I'm...

```
update cf_ctuser_roles set DESCRIPTION='"Good student" basics. Manipulate most things at SpecimenDetail; manage Citations' where ROLE_NAME='manage_specimens';
	
grant insert,update,delete on citation to manage_specimens;
```

... and closing this Issue; please re-open if further discussion is required.

**Comment by jldunnum** [comment]
Not sure I agree that we need to restrict students from adding publications.This is something I would actually like to have students do.  It is pretty straight forward now in terms of grabbing a DOI and saving it.  

On this topic, but maybe a seperate issue: We have the issue of the authors of the added papers not automatically being added as agents.  Noone has time to individually add each author for every paper as an agent.  However, if you don't, you can't find a publication through an author search.  Maybe we should require that at least the first author of a pub be an agent and make an agent add button in the pub creation screen so you don't have to back out and create the agent and then go back into the pub to select it again.

**Comment by dustymc** [comment]
>students ... publications

Nothing is stopping you from granting them that access - but maybe something (no idea what that might be - I still like Gordon's written test idea...) should be since with it they can update "MVZ publications" (which are of course not actually "owned" by anyone, and may be "MVZ's" only because you haven't added MSB citations yet).

>authors of the added papers not automatically being added

That's true - I don't have THAT much confidence in my scripts! - but when I can guess at them they're one click (each) away. I pasted a DOI in, clicked crossref, and got...

<img width="703" alt="screen shot 2016-06-07 at 1 56 19 pm" src="https://cloud.githubusercontent.com/assets/5720791/15874437/0410327e-2cb8-11e6-8c0b-5ca151a56814.png">

If anyone wants to do anything useful with authors it's definitely important that those links get clicked before saving, at least for the important authors. (And if my scripts can guess correctly it's pretty likely that the author is important, even if only to exclude incorrect guesses.) Do we need better documentation somewhere, or those suggestions to be giant red font, or ????

>Maybe we should require

We used to require all publication authors be entered as agents, but it was at least sometimes a LOT of work for not much obvious benefit - a paper would have 30 authors, the 17th being the only person to whom we'd loaned specimens and the rest people we've never heard of before and not likely to hear from again. I _think_ we'd get a consensus that globally requiring anything about authors is not a great use of resources, but certainly open a new issue if you think otherwise. And of course you can instruct whoever's entering your publications to do whatever you want.

>make an agent add button in the pub creation screen

File a new Issue - we've at various times played with adding agents from the agent picker, and the scripts that keep agents clean are probably smart enough to have another go at that. #732 needs resolved first - we currently have two agent picks, and I don't really want to write/maintain code to both of them.

**Comment by dustymc** [comment]
CONSENSUS @ AWG meeting 1 Aug 2016: move citations to manage_specimens role

**Comment by dustymc** [comment]
ADDITION: disallow save unless at least one agent

**Comment by mkoo** [comment]
Consensus: At least add in lead author as agent and linking in Arctos for discoverability. Then add in as many as already agents in Arctos as an option for individual curators.

Suggestion to have a link to the Add Agent button on the Add Publication page (maybe as Overlay instead of pop-ups #732)
